### ISSUE TITLE
[HOLD 19496][$1000]  Does wrong navigation to screen on back in 2FA enable

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#21451]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Action Performed:
1. Open Setting
2. Tap on security
3. Tap on  Two-factor authentication
4. copy code and click on next
5. get code from authenticate app and enter code and click on next
6. click on Got it
7. click on back
## Expected Result: 
Redirect directly in setting screen  after authentication was enable on back press.
## Actual Result:
 first time is show back screens.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 
**Reproducible in staging?:** needs reproduction
**Reproducible in production?:** needs reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
https://github.com/Expensify/App/assets/43996225/920e3cea-fafb-41a2-843f-427f734d785a

**Expensify/Expensify Issue URL:**
**Issue reported by:** @jaydipramani
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1685349692929619

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01646f8c682cf72533</li>
        <li>Upwork Job ID: 1674398083212304384</li>
        <li>Last Price Increase: 2023-06-29</li>
    </ul>
</details>

**Comment by Shivam7-1** [comment]
## Proposal

### Does wrong navigation to screen on back in 2FA enable

## Please re-state the problem that we are trying to solve in this issue.
What is the root cause of that problem?
In the `useEffect` hook, there is a condition that checks for the visibility of the form using the `this.props.isVisible` prop. When the form becomes visible (`this.props.isVisible` is `true`), the `this.validateCode` state is being cleared. This causes the form to reset to an empty state whenever it becomes visible.

The issue is lies on this below code in 
https://github.com/Expensify/App/blob/main/src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.js (89-102 line )

`useEffect(() => {

        if (!inputValidateCodeRef.current) {

            return;
        }

        if (!validateCode) {
            return;
        }
        // Clear the code input if magic code valid
        if (prevIsVisible && !props.isVisible) {
            setValidateCode('');
        }
    }, [props.isVisible, prevIsVisible, validateCode]);`

### What changes do you think we should make in order to solve the problem?
In Solution of this issue I will add additional condition to check whether `validateCode` is not empty before clearing it. This way, the code input will only be cleared if it has a value, allowing the form to retain the entered code if it was not empty.

By making this modification, the form will still reset the code input when becoming invisible, but it will preserve the code if it was entered by the user. And it will won't disturb `UseEffect` Function

**Comment by Shivam7-1** [comment]
@CortneyOfstad  Can you please Review My Above Proposal Regarding This Issue

**Comment by CortneyOfstad** [comment]
@kavimuru is there any video or screenshots showing this issue, so I can confirm the steps that were taken?

**Comment by CortneyOfstad** [comment]
Thanks @kavimuru and sorry for missing that! I was able to recreate this, so going to get eyes on this 👍

**Comment by Shivam7-1** [comment]
## Proposal

### Does wrong navigation to screen on back in 2FA enable

## Please re-state the problem that we are trying to solve in this issue.
What is the root cause of that problem?
In the `useEffect` hook, there is a condition that checks for the visibility of the form using the `this.props.isVisible` prop. When the form becomes visible (`this.props.isVisible` is `true`), the `this.validateCode` state is being cleared. This causes the form to reset to an empty state whenever it becomes visible.

The issue is lies on this below code in 
https://github.com/Expensify/App/blob/main/src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.js (89-102 line )

`useEffect(() => {

        if (!inputValidateCodeRef.current) {

            return;
        }

        if (!validateCode) {
            return;
        }
        // Clear the code input if magic code valid
        if (prevIsVisible && !props.isVisible) {
            setValidateCode('');
        }
    }, [props.isVisible, prevIsVisible, validateCode]);`

### What changes do you think we should make in order to solve the problem?
In Solution of this issue I will add additional condition to check whether `validateCode` is not empty before clearing it. This way, the code input will only be cleared if it has a value, allowing the form to retain the entered code if it was not empty.

By making this modification, the form will still reset the code input when becoming invisible, but it will preserve the code if it was entered by the user. And it will won't disturb `UseEffect` Function

**Comment by koko57** [comment]
Shouldn't it be put on hold for the https://github.com/Expensify/App/issues/19496?

**Comment by CortneyOfstad** [comment]
@koko57 Good catch — @mananjadhav do you think the same? If so, I'll change the frequency of this to weekly instead of daily, so we can keep an eye out 👍

**Comment by mananjadhav** [comment]
I am not sure if we should put it on hold. But I’ll take a look at the issues and proposals by tomorrow and respond. Thanks for the patience here.

**Comment by CortneyOfstad** [comment]
Sounds good — I am heading OoO for the week, so going to re-assign in the meantime 👍

**Comment by Christinadobrzyn** [comment]
@mananjadhav can you provide any insight into if this should be on hold - when you have a moment!

**Comment by mananjadhav** [comment]
@Christinadobrzyn @CortneyOfstad Went through the issue, and the linked issue, I think it makes sense to put it on Hold.

**Comment by Christinadobrzyn** [comment]
Oh hold for - https://github.com/Expensify/App/issues/19496

**Comment by Christinadobrzyn** [comment]
I'm going to be ooo until August 31st so going to unassign and assign a new teammate. 

At this time, this is on hold for - https://github.com/Expensify/App/issues/19496

**Comment by laurenreidexpensify** [comment]
Still on hold for https://github.com/Expensify/App/issues/19496

**Comment by Christinadobrzyn** [comment]
I'm back from ooo - thanks for handling this @laurenreidexpensify - happy to take it off your plate

**Comment by Christinadobrzyn** [comment]
Still on hold for https://github.com/Expensify/App/issues/19496

**Comment by Christinadobrzyn** [comment]
Still on hold for https://github.com/Expensify/App/issues/19496

**Comment by Christinadobrzyn** [comment]
Still on hold for https://github.com/Expensify/App/issues/19496

**Comment by Christinadobrzyn** [comment]
Payment on https://github.com/Expensify/App/issues/19496 soon!

**Comment by Christinadobrzyn** [comment]
I think we're waiting on proposals for this. @mananjadhav let me know if you think differently!

**Comment by Christinadobrzyn** [comment]
Hey @mananjadhav do you think we can reach out to Software Mansion or otherwise to pick up this job? Or should I raise the bounty? It's been sitting for a while!

**Comment by Christinadobrzyn** [comment]
nudge @mananjadhav

**Comment by mananjadhav** [comment]
I checked and I am not able to reproduce this. Are you still able to ? I checked on Web, Desktop and Android Native?

**Comment by Christinadobrzyn** [comment]
Ah yes, you are right @mananjadhav this is no longer an issue. I think this can be closed. 

I think we need to pay the report- should we be paying anyone else?

**Comment by mananjadhav** [comment]
@Christinadobrzyn I would like to request for a partial payout for the review of the proposals. Let me know if that works.

Yes we need to payout for the reporting.

**Comment by Christinadobrzyn** [comment]
Hey @mananjadhav what would be a reasonable payment for the work you did on this?

**Comment by mananjadhav** [comment]
I think 25% of the job amount. Basically I am following the guidelines based on this [document](https://docs.google.com/document/d/1t9KDPiEJmTEIVpo7hUsAGqdDdazIHY2A6znewEf5S4s/edit#heading=h.198et3u52rxj).

**Comment by Christinadobrzyn** [comment]
Okay, I think $250 is reasonable for the work on this! 

I sent a job invite to you - https://www.upwork.com/jobs/~01981c4c0707d6b1b3
Internal posting - https://www.upwork.com/ab/applicants/1706763284355817472/job-details

**Comment by mananjadhav** [comment]
@Christinadobrzyn I am paid via ND. Can you just post the payment summary? I'll raise the request on NewDot.

**Comment by jaydipramani** [comment]
@Christinadobrzyn how to get report bonus?

**Comment by Christinadobrzyn** [comment]
Sorry @jaydipramani I should've also asked for your Upwork profile yesterday. Can you send that over so I can pay you as the reporter? Thanks!

Payouts due:

Issue Reporter: $250 @jaydipramani 
Contributor: NA
Contributor+: $250 @mananjadhav 

Eligible for 50% #urgency bonus? N

Upwork job is [here](https://www.upwork.com/jobs/~01981c4c0707d6b1b3).

**Comment by jaydipramani** [comment]
@Christinadobrzyn 
Upwork profile:
https://www.upwork.com/freelancers/~017e5c3edbd219071d

**Comment by Christinadobrzyn** [comment]
Thanks @jaydipramani can you accept my offer to the job and I'll pay this? https://www.upwork.com/jobs/~01981c4c0707d6b1b3

**Comment by jaydipramani** [comment]
@Christinadobrzyn  Accepted 👍

**Comment by Christinadobrzyn** [comment]
Thanks @jaydipramani I paid you in Upwork. 

Closing this GH. Let me know if you need anything else!

**Comment by JmillsExpensify** [comment]
$250 payment approved for @mananjadhav based on BZ summary.
